File: src/main/java/com/example/sneakerx/dtos/cart/CartItemResponse.java
package com.example.sneakerx.dtos.cart;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CartItemResponse {
    // --- 1. Cart Management ---
    private Integer itemId;       // To update/delete this specific row
    private Integer skuId;        // To identify the exact variant
    private Integer quantity;     // Current quantity in cart

    // --- 2. Product Navigation & Display ---
    private Integer productId;    // To navigate back to Product Detail Screen
    private String productName;   // e.g., "Nike Air Force 1"
    private String shopName;      // e.g., "Nike Official Store" (Good for multi-vendor grouping)
    private Integer shopId;       // To group items by Shop in checkout

    // --- 3. Visuals ---
    private String productImageUrl; // The specific variant image (Red Shoe) or fallback to Main Image

    // --- 4. Variant Info (Crucial) ---
    private String skuCode;       // e.g., "NK-AF1-RED-42"
    private String variantDescription; // e.g., "Color: Red, Size: 42" (Pre-formatted string)

    // --- 5. Financials ---
    private Double unitPrice;     // The current price of the SKU
    private Double subTotal;      // unitPrice * quantity (Backend should calculate this)

    // --- 6. Stock Validation (UX) ---
    private Integer currentStock; // To show "Only 2 left!" or disable "+" button
    private boolean available;  // True if stock >= quantity
}

File: lib/src/modules/cart/dtos/cart_item_response.dart
class CartItemResponse {
  final int itemId;
  final int skuId;
  final int quantity;
  final int productId;
  final String productName;
  final String shopName;
  final int shopId;
  final String productImageUrl;
  final String skuCode;
  final String variantDescription;
  final double unitPrice;
  final double subTotal;
  final int currentStock;
  final bool available;

  CartItemResponse({
    required this.itemId,
    required this.skuId,
    required this.quantity,
    required this.productId,
    required this.productName,
    required this.shopName,
    required this.shopId,
    required this.productImageUrl,
    required this.skuCode,
    required this.variantDescription,
    required this.unitPrice,
    required this.subTotal,
    required this.currentStock,
    required this.available,
  });

  factory CartItemResponse.fromJson(Map<String, dynamic> json) {
    return CartItemResponse(
      itemId: json['itemId'] ?? 0,
      skuId: json['skuId'] ?? 0,
      quantity: json['quantity'] ?? 0,
      productId: json['productId'] ?? 0,

      // Strings: Default to empty if null
      productName: json['productName'] ?? "Unknown Product",
      shopName: json['shopName'] ?? "Unknown Shop",
      shopId: json['shopId'] ?? 0,
      productImageUrl: json['productImageUrl'] ?? "",

      // Variant Info
      skuCode: json['skuCode'] ?? "",
      variantDescription: json['variantDescription'] ?? "",

      // Financials: Use 'num' to safely handle int/double mismatch from API
      unitPrice: (json['unitPrice'] as num?)?.toDouble() ?? 0.0,
      subTotal: (json['subTotal'] as num?)?.toDouble() ?? 0.0,

      // Stock & Logic
      currentStock: json['currentStock'] ?? 0,
      available: json['available'] ?? false,
    );
  }
}